Cache/IO.lean

```diff
@@ -425,16 +425,16 @@ def lookup (hashMap : ModuleHashMap) (modules : List Name) : IO Unit := do
 
 /--
 Parse a string as either a path or a Lean module name.
-TODO: If the argument describes a folder, use `walkDir` to find all `.lean` files within.
+If the argument describes a folder, use `walkDir` to find all `.lean` files within.
 
-Return tuples of the form ("module name", "path to .lean file").
+Return tuples of the form `("module name", "path to .lean file")`.
 
 The input string `arg` takes one of the following forms:
 
 1. `Mathlib.Algebra.Fields.Basic`: there exists such a Lean file
-2. `Mathlib.Algebra.Fields`: no Lean file exists but a folder (TODO)
+2. `Mathlib.Algebra.Fields`: no Lean file exists but a folder
```

Comment:
I think we should match lake here, and use `.+` syntax. Otherwise we're stuck if both `Foo/Bar.lean` and `Foo/Bar/Baz.lean` exist.

We could also support `.*` syntax, which would mean "load both of the above", though of course the user has to remember to escape it in some shells.